js_modules/dagster-ui/packages/ui-core/src/search/useGlobalSearch.tsx

```diff
@@ -207,7 +297,7 @@ export const useGlobalSearch = () => {
     loading: secondaryDataLoading,
   } = useIndexedDBCachedQuery<SearchSecondaryQuery, SearchSecondaryQueryVariables>({
     query: SEARCH_SECONDARY_QUERY,
-    key: 'SearchSecondary',
+    key: `SearchSecondary:${SEARCH_SECONDARY_DATA_VERSION}`,
```

Comment:
This works but it leaves the old cache in place so it never gets cleaned up. I think instead `useIndexedDBCachedQuery` should take a third parameter `version` that can be used to tell `useIndexedDBCacheQuery` to ignore caches on a different version.

Reply:
Makes sense. This PR is updated to now also store the data version in the cache.

I added a data version for both the primary and secondary queries since we should always be providing a version